processor/schemaprocessor/internal/translation/version.go

```diff
@@ -0,0 +1,169 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+)
+
+const separator = "."
+
+var (
+	ErrInvalidFamily  = errors.New("invalid schema family")
+	ErrInvalidVersion = errors.New("invalid schema version")
+)
+
+//	The following status values define whether the transformer
+//	has to revert changes or update changes to the signal being modified
+//	These values match the Version.Compare out when performing:
+//		from.Compare(to) // ie: (1.0.0).Compare(1.2.1)
+const (
+	Update   int = -1 // From is less than To
+	NoChange int = 0  // From equals To
+	Revert   int = 1  // From is greater than To
+)
+
+// Version is a machine readable version of the string
+// schema identifier that can assist in making indexing easier
+type Version struct {
+	Major, Minor, Patch int
+}
+
+// ReadVersionFromPath allows for parsing paths
+// that end in a schema indentifier
+func ReadVersionFromPath(p string) (*Version, error) {
+	if p == "" {
+		return nil, fmt.Errorf("empty path:%w", ErrInvalidVersion)
+	}
+	// path.Base doesn't enforce strict forward slashes
+	// thus having to test for it here
+	if strings.HasSuffix(p, "/") {
+		return nil, fmt.Errorf("must not have trailing slash: %w", ErrInvalidVersion)
+	}
+	ident := path.Base(p)
+	return NewVersion(ident)
+}
+
+// GetFamilyAndVersion takes a schemaURL and separates the family from the identifier.
+func GetFamilyAndVersion(schemaURL string) (family string, id *Version, err error) {
+	u, err := url.Parse(schemaURL)
+	if err != nil {
+		return "", nil, err
+	}
+	id, err = ReadVersionFromPath(u.Path)
+	if err != nil {
+		return "", nil, err
+	}
+
+	u.Path = path.Dir(u.Path)
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return "", nil, fmt.Errorf("must use http(s): %w", ErrInvalidFamily)
+	}
+	if u.Host == "" {
+		return "", nil, fmt.Errorf("must have a host name: %w", ErrInvalidFamily)
+	}
+
+	return u.String(), id, err
+}
+
+func joinSchemaFamilyAndVersion(family string, version *Version) string {
```

Comment:
This is unused so unclear how it will be used. Also not quite clear what it actually does. Does it only use the "path" option of the URL? If so the func name seems not completely accurate. Perhaps delete from this PR and  introduce when it is needed.

Reply:
This is used in a future PR but I forgot to clean it up when I split up the big complete implementation.